### DISCUSSION TITLE
Multiple "type" within "series"

### DISCUSSION THREAD

**Discussion by humphreylee** [yuankunzhang/charming#125] [Q&A] [answered]
I’ve observed that ECharts allows for multiple “types” within a “series” in examples like this [one](https://echarts.apache.org/examples/en/editor.html?c=scatter-aqi-color). Additionally, different “types” can be mixed within a single series, as shown in this [example](https://echarts.apache.org/examples/en/editor.html?c=scatter-exponential-regression). I believe there are benefits to this approach. However, I’ve tried implementing this in Charming without success so far. Is it possible to achieve this in Charming?

**Comment by LukaOber** [comment]
I don't understand what you mean. In both of your examples, each series only has one type.
You can have multiple series in charming by repeatedly calling `.series()`

First example, there are three scatter series.
```rust
Chart::new()
  .series(Scatter::new())
  .series(Scatter::new())
  .series(Scatter::new());
```
Second example, one scatter series and one line series.
```rust
Chart::new()
  .series(Scatter::new())
  .series(Line::new());
```

**Comment by humphreylee** [comment]
Currently, the approach involves calling ```.series()``` for each series individually. I may have misunderstood, thinking it could be translated as follows:
```
Chart::new()
  .series(
        Scatter::new(), 
        Line::new()
  );
```
This [example ](https://echarts.apache.org/examples/en/editor.html?c=scatter-exponential-regression) includes the following code snippet, suggesting an array/vector/tupple of series types, which also can share common properties, such as dataset.
```
series: [
    {
      name: 'scatter',
      type: 'scatter',
      datasetIndex: 0
    },
    {
      name: 'line',
      type: 'line',
      smooth: true,
      datasetIndex: 1,
      symbolSize: 0.1,
      symbol: 'circle',
      label: { show: true, fontSize: 16 },
      labelLayout: { dx: -20 },
      encode: { label: 2, tooltip: 1 }
    }
  ]
```

**Reply by LukaOber** [reply]
These are two separate series, and `dataset` is set on `Chart`. (I think)

**Comment by humphreylee** [comment]
It is unequivocal that these are two distinct series. The ```.series()``` function (as a [function](https://github.com/yuankunzhang/charming/blob/main/charming/src/lib.rs)) in Charming may present a limitation, in contrast to ECharts, which utilizes a [JSON configuration format](https://www.gislite.com/tutorial/k8228).”

There is a comparable ECharts wrapper available in the Go language ([go-echarts](https://github.com/go-echarts/go-echarts)), which saw early [contributions](https://github.com/go-echarts/go-echarts/commits/master/?after=d386dfbeba8e4e0b6b0ebe0588f0cf86eb5ef736+384) from [Zhang](https://github.com/yuankunzhang), that functions in a similar manner.

**Reply by LukaOber** [reply, marked as answer]
I don't think a json configuration is better. They are different approaches. The json format is way simpler to implement, offers basically all options but the implementation here provides type safety. It's a tradeoff.

**Reply by humphreylee** [reply]
> the implementation here provides type safety
👍
